ebean-core/src/main/java/io/ebeaninternal/server/core/DefaultServer.java

```diff
@@ -630,7 +626,11 @@ private <T> T referenceFindOne(Class<T> type, Object id, BeanDescriptor<?> desc)
     }
     // we actually need to do a query because we don't know the type without the discriminator
     // value, just select the id property and discriminator column (auto added)
-    return find(type).select(idProp.name()).setId(id).findOne();
+    T bean = find(type).select(idProp.name()).setId(id).findOne();
+    if (bean == null) {
+      throw new NullPointerException("Could not find reference bean " + id + " for " + desc);
+    }
```

Comment:
Code change: I throw an NPE now.

Reply:
NPE is more like for preconditions or if you really messed up your code, don't you think? How does this exception propagate? I would not expect to received a NPE in user code if an entity isn't found. Isn't there an EntityNotFoundException?

Reply:
yes, thanks. EntityNotFoundException is the better choice here

Reply:
Behaviour change means it should be in its own PR.  We don't want subtle behaviour changes in a PR that says "Improved Null-checking annotations"  and we do want behaviour changes in their own PR that justifies why the change is good.